types/topology.go

```diff
@@ -10,14 +10,16 @@ type Topology struct {
 	Defaults *NodeDefinition            `yaml:"defaults,omitempty"`
 	Kinds    map[string]*NodeDefinition `yaml:"kinds,omitempty"`
 	Nodes    map[string]*NodeDefinition `yaml:"nodes,omitempty"`
-	Links    []*LinkConfig              `yaml:"links,omitempty"`
+	RawLinks []*RawLinkType             `yaml:"links,omitempty"`
+	Links    []*LinkConfig              `yaml:"-"`
```

Comment:
it seems to me that `Links` in its current form shouldn't be part of the Topology struct

What if you make it like that:

```go
    Links []*LinkDefinition             `yaml:"links,omitempty"`
```
where `LinkDefinition` is what is your RawLinkType is currently?

Reply:
yes, fine.

Reply:
Take a look at [974872c](https://github.com/srl-labs/containerlab/pull/1450/commits/974872c488388b6757a39e3d30acfc88a3e27384) where this is addressed.